### ISSUE TITLE
Recommendations

### ISSUE THREAD

**Issue by fvictorio** [aurora-is-near/hardhat-storage-layout#4]
Hi, I have some suggestions for this plugin in case you're interested:

1. Instead of making the user do `await hre.storageLayout.export()` in a script, create a task for that. It's probably just a matter of doing this:

    ```js
    task("storage-layout", async (taskArgs, hre) => {
      await hre.storageLayout.export();
    })
    ```

    And then users can just run `hh storage-layout`

2. In line with the previous item, maybe it's worth adding an optional argument to select the contract that will be printed, defaulting to printing all of them if no argument is passed.
3. Finally, I think you could avoid making users configure the compiler by overriding the `TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE` subtask to change the compiler settings. The downside of this approach is that that setting will be enabled for all compilations, but that's probably fine. (You could also override it only when the `storage-layout` task is used, but that's a bit more involved and has other trade-offs.)

Let me know if you have questions about any of these items!

**Comment by Leon-Africa** [comment]
I like 1 and 2!

To reference: https://github.com/aurora-is-near/hardhat-storage-layout/issues/6#issue-957038338

````hh storage-layout --outputdir myoutputdir````

**Comment by qzirak** [comment]
@fvictorio Is there any update on this ?

**Comment by fvictorio** [comment]
@qzirak I'm not one of the plugin's authors, I was just suggesting some potential improvements :slightly_smiling_face:

**Comment by qzirak** [comment]
@fvictorio Yeah, I was asking about the suggested update. For me it does not work. I get an error of 'storage of undefined' for some reason.
